js/src/ui/TxList/TxRow/txRow.js

```diff
@@ -137,11 +149,113 @@ class TxRow extends Component {
     return (
       <td className={ styles.timestamp }>
         <div>{ blockNumber && block ? moment(block.timestamp).fromNow() : null }</div>
-        <div>{ blockNumber ? _blockNumber.toFormat() : 'Pending' }</div>
+        <div>{ blockNumber ? _blockNumber.toFormat() : this.renderCancelToggle() }</div>
       </td>
     );
   }
 
+  renderCancelToggle () {
+    const { canceled, editing, isCancelOpen, isEditOpen } = this.state;
+
+    if (canceled) {
+      return (
+        <div className={ styles.pending }>
+          <FormattedMessage
+            id='ui.txList.txRow.canceled'
+            defaultMessage='CANCELED'
+          />
+        </div>
+      );
+    }
+
+    if (editing) {
+      return (
+        <div className={ styles.pending }>
+          <FormattedMessage
+            id='ui.txList.txRow.editing'
+            defaultMessage='EDITING'
+          />
+        </div>
+      );
+    }
+
+    if (!isCancelOpen && !isEditOpen) {
+      const pendingStatus = this.getCondition();
+
+      if (pendingStatus === 'submitting...') {
+        return (
+          <div className={ styles.pending }>
+            <div />
+            <div>
+              <FormattedMessage
+                id='ui.txList.txRow.submitting'
+                defaultMessage='SUBMITTING'
+              />
+            </div>
+          </div>
+        );
+      }
+      return (
+        <div className={ styles.pending }>
+          <span>
+            <FormattedMessage
+              id='ui.txList.txRow.time'
+              defaultMessage='{ which }'
+              values={ { which: pendingStatus } }
+            />
+          </span>
+          <div>
+            <FormattedMessage
+              id='ui.txList.txRow.scheduled'
+              defaultMessage='SCHEDULED'
+            />
+          </div>
+          <a onClick={ this.setEdit }>
+            <FormattedMessage
+              id='ui.txList.txRow.edit'
+              defaultMessage='EDIT'
+            />
+          </a>
+          <span>{' | '}</span>
+          <a onClick={ this.setCancel }>
+            <FormattedMessage
+              id='ui.txList.txRow.cancel'
+              defaultMessage='CANCEL'
+            />
+          </a>
+        </div>
+      );
+    }
+
+    return (
+      <div className={ styles.pending }>
+        <div />
+        <div>
+          <FormattedMessage
+            id='ui.txList.txRow.verify'
+            defaultMessage='ARE YOU SURE?'
+          />
+        </div>
+        <a onClick={ (isCancelOpen) ? this.cancelTx : this.editTx }>
+          <FormattedMessage
+            id='ui.txList.txRow.verify.cancelEdit'
+            defaultMessage='{ which }'
+            values={ {
+              which: `${(isCancelOpen) ? 'Cancel' : 'Edit'}`
+            } }
+          />
+        </a>
+        <span>{' | '}</span>
+        <a onClick={ this.nevermind }>
```

Comment:
Would name this something else, e.g. not `nevermind` (`revertEdit` or `cancelEdit` or ...)